src/webapi/window.rs

```diff
@@ -86,4 +106,25 @@ impl Window {
             ).into_reference_unchecked()
         }
     }
+
+    /// You should call this method whenever you're ready to update your animation onscreen.
+    /// This will request that your animation function be called before the browser performs the next repaint.
+    /// The number of callbacks is usually 60 times per second, but will generally match the display refresh 
+    /// rate in most web browsers as per W3C recommendation. request_animation_frame() calls are paused in most browsers
+    /// when running in background tabs or hidden <iframe>s in order to improve performance and battery life.
+    ///
+    /// The callback method is passed a single argument, a f64, which indicates the current time when
+    /// callbacks queued by requestAnimationFrame() begin to fire. Multiple callbacks in a single frame, therefore,
+    /// each receive the same timestamp even though time has passed during the computation of every previous callback's workload.
+    /// This timestamp is a decimal number, in milliseconds, but with a minimal precision of 1ms (1000 µs).
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame)
+    pub fn request_animation_frame< F: FnOnce(f64) + 'static>( &self, callback: F) -> RequestAnimationFrameHandle {
+        let values: Value = js!{
+            var callback = @{Once(callback)};
+            var request = @{self}.requestAnimationFrame(callback);
+            return { request, callback };
```

Comment:
Returning an array here would probably be very slightly more efficient instead of an object. Maybe? (Honestly I don't know!) Either use an array, or an object initialized the old way. (This syntax is ES6+ only.)

Reply:
objects and array are just syntactic sugar a lot of times in javascript. They end up getting stored the same a lot of the time. I just did it this way to make it clearer when I used the values later. I'll change it to the es5 way